services/rows/src/rows/routes/rows.py

```diff
@@ -123,29 +116,10 @@ async def rows_endpoint(request: Request) -> Response:
                         )
                 with StepProfiler(method="rows_endpoint", step="query the rows"):
                     pa_table = rows_index.query(offset=offset, length=length)
-                with StepProfiler(method="rows_endpoint", step="clean cache"):
-                    # no need to do it every time
-                    if random.random() < clean_cache_proba:  # nosec
-                        if (
-                            keep_first_rows_number < 0
-                            and keep_most_recent_rows_number < 0
-                            and max_cleaned_rows_number < 0
-                        ):
-                            logger.debug(
-                                "Params keep_first_rows_number, keep_most_recent_rows_number and"
-                                " max_cleaned_rows_number are not set. Skipping cached assets cleaning."
-                            )
-                        else:
-                            clean_cached_assets(
-                                dataset=dataset,
-                                cached_assets_directory=cached_assets_directory,
-                                keep_first_rows_number=keep_first_rows_number,
-                                keep_most_recent_rows_number=keep_most_recent_rows_number,
-                                max_cleaned_rows_number=max_cleaned_rows_number,
-                            )
                 with StepProfiler(method="rows_endpoint", step="transform to a list"):
                     response = create_response(
                         dataset=dataset,
+                        revision=revision,  # type: ignore
```

Comment:
why the "type: ignore"? It feels weird: if it can be None, we should generate an error instead

Reply:
Makes sense. In this case, we use an Optional revision (initialized at the beginning of the function) in case of error but since we use `create_response` only when `revision = rows_index.revision` is success I think we don't need other validation (Since dataset_git_revision is mandatory now and it is read from the cache in https://github.com/huggingface/datasets-server/blob/main/libs/libcommon/src/libcommon/parquet_utils.py#L290
`mypy` was alerting because we had `try_backfill_dataset_then_raise` block before calling  `create_response` (that is why it didn't fail for /search and /filter) but I fixed the order of execution.